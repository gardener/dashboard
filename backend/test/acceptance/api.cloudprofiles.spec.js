//
// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { mockRequest } = require('@gardener-dashboard/request')

describe('api', function () {
  let agent

  beforeAll(() => {
    agent = createAgent()
  })

  afterAll(() => {
    return agent.close()
  })

  beforeEach(() => {
    mockRequest.mockReset()
  })

  describe('cloudprofiles', function () {
    const user = fixtures.user.create({ id: 'john.doe@example.org' })

    it('should return all cloudprofiles', async function () {
      mockRequest.mockImplementationOnce(fixtures.auth.mocks.reviewSelfSubjectAccess())

      const res = await agent
        .get('/api/cloudprofiles')
        .set('cookie', await user.cookie)
        .expect('content-type', /json/)

      expect(mockRequest).toBeCalledTimes(1)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })
  })
})
