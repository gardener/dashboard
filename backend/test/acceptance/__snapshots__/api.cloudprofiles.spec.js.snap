// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api cloudprofiles should return all cloudprofiles 1`] = `
Array [
  Array [
    Object {
      ":authority": "kubernetes:6443",
      ":method": "post",
      ":path": "/apis/authorization.k8s.io/v1/selfsubjectaccessreviews",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImpvaG4uZG9lQGV4YW1wbGUub3JnIiwiaWF0IjoxNTc3ODM2ODAwLCJhdWQiOlsiZ2FyZGVuZXIiXSwiZXhwIjozMTU1NzE2ODAwLCJqdGkiOiJqdGkifQ.LkQ9PEN893UNTsZZn2Ux_CAYNOoQ2ISboWuHiAc5HHU",
    },
    Object {
      "apiVersion": "authorization.k8s.io/v1",
      "kind": "SelfSubjectAccessReview",
      "spec": Object {
        "nonResourceAttributes": undefined,
        "resourceAttributes": Object {
          "group": "core.gardener.cloud",
          "resource": "cloudprofiles",
          "verb": "list",
        },
      },
    },
  ],
]
`;

exports[`api cloudprofiles should return all cloudprofiles 2`] = `
Array [
  Object {
    "data": Object {
      "kubernetes": Object {
        "versions": Array [
          Object {
            "version": "1.9.0",
          },
          Object {
            "version": "1.8.5",
          },
        ],
      },
      "seedNames": Array [
        "infra1-seed",
        "infra1-seed2",
        "infra4-seed-without-secretRef",
      ],
      "seedSelector": Object {},
      "type": "infra1",
    },
    "metadata": Object {
      "cloudProviderKind": "infra1",
      "displayName": "infra1-profileName",
      "name": "infra1-profileName",
    },
  },
  Object {
    "data": Object {
      "kubernetes": Object {
        "versions": Array [
          Object {
            "version": "1.9.0",
          },
          Object {
            "version": "1.8.5",
          },
        ],
      },
      "seedNames": Array [
        "infra3-seed",
        "infra3-seed-with-selector",
      ],
      "seedSelector": Object {
        "providerTypes": Array [
          "infra2",
          "infra3",
        ],
      },
      "type": "infra1",
    },
    "metadata": Object {
      "cloudProviderKind": "infra1",
      "displayName": "infra1-profileName2",
      "name": "infra1-profileName2",
    },
  },
  Object {
    "data": Object {
      "kubernetes": Object {
        "versions": Array [
          Object {
            "version": "1.9.0",
          },
          Object {
            "version": "1.8.5",
          },
        ],
      },
      "seedNames": Array [
        "infra3-seed-with-selector",
      ],
      "seedSelector": Object {
        "matchLabels": Object {
          "foo": "bar",
        },
      },
      "type": "infra3",
    },
    "metadata": Object {
      "cloudProviderKind": "infra3",
      "displayName": "infra3-profileName",
      "name": "infra3-profileName",
    },
  },
  Object {
    "data": Object {
      "kubernetes": Object {
        "versions": Array [
          Object {
            "version": "1.9.0",
          },
          Object {
            "version": "1.8.5",
          },
        ],
      },
      "seedNames": Array [
        "infra3-seed",
        "infra3-seed-with-selector",
      ],
      "seedSelector": Object {},
      "type": "infra3",
    },
    "metadata": Object {
      "cloudProviderKind": "infra3",
      "displayName": "infra3-profileName2",
      "name": "infra3-profileName2",
    },
  },
]
`;
